SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp

```diff
@@ -193,3 +196,114 @@ void USpatialNetConnection::OnHeartbeat()
 {
 	SetHeartbeatTimeoutTimer();
 }
+
+void USpatialNetConnection::InitPing(FTimerManager* InTimerManager, Worker_EntityId InPlayerControllerEntity)
+{
+	checkf(PlayerControllerEntity == SpatialConstants::INVALID_ENTITY_ID || PlayerControllerEntity == InPlayerControllerEntity, TEXT("InitPing: PlayerControllerEntity already set: %lld. New entity: %lld"), PlayerControllerEntity, InPlayerControllerEntity);
+	PlayerControllerEntity = InPlayerControllerEntity;
+	TimerManager = TimerManager == nullptr ? InTimerManager : TimerManager;
+
+	TWeakObjectPtr<USpatialNetConnection> ConnectionPtr = this;
+
+	if (Driver->IsServer())
+	{
+		Cast<USpatialNetDriver>(Driver)->Receiver->AddClientPongDelegate(PlayerControllerEntity, ClientPongDelegate::CreateLambda([ConnectionPtr](Worker_ComponentUpdateOp& Op)
+		{
+			if (ConnectionPtr.IsValid())
```

Comment:
We should flip conditionals like this around. There would benefit from less indentation.